### PR TITLE
battery: reset current filter when transitioning to FW

### PR DESCRIPTION
**Pull request by sfuhrer**
VTOLs consume a lot more power in hover compared to fixed-wing flight. The remaining flight time thus should reset if one switches from MC to FW, as otherwise it takes several minutes until the estimate goes down.

### Changelog Entry
For release notes:
```
Improvement: battery: reset current filter when transitioning to FW
```

### Alternatives
Only update the filter in FW for VTOL. As VTOLs though also can have full hover flights this is for me the worse option.

### CODE REVIEW

**Review by tstastny — APPROVED** [review]

**Comment by tstastny** [comment]
@sfuhrer i think this is another example of where we need to split things up -- this is another "if VTOL" case where probably it's better to consider battery estimation in an airframe specific way. could we spend a little time thinking of follow-on to this, how we want to detangle?

for short term.. this will work.

**Comment by sfuhrer** [comment]
Rebased on current main

**Comment by sfuhrer** [comment]
Flight tested: 
image

File: src/lib/battery/battery.cpp [hunk 1]
```diff
@@ -305,19 +305,30 @@ void Battery::computeScale()
...
+	bool reset_current_avg_filter = false;
 
 	if (_vehicle_status_sub.updated()) {
 		vehicle_status_s vehicle_status;
 
 		if (_vehicle_status_sub.copy(&vehicle_status)) {
 			_armed = (vehicle_status.arming_state == vehicle_status_s::ARMING_STATE_ARMED);
 			_vehicle_status_is_fw = (vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING);
+
+			if (vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING && _vtol_was_in_mc_mode) {
```
Comment (bresch) [inline src/lib/battery/battery.cpp:317]: You have `_vehicle_status_is_fw` already, so you don't need an additional `_vtol_was_in_mc_mode`, just use it and update the variable after.
```
if (vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING && !_vehicle_status_is_fw) {
...
}

_vehicle_status_is_fw = (vehicle_status.vehicle_type == vehicle_status_s::VEHICLE_TYPE_FIXED_WING);
```

**Review by bresch — CHANGES_REQUESTED** [review]

File: src/lib/battery/battery.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/lib/battery/battery.cpp]: very good point, I missed that we now have this other class variable already. 

Addressed it.

**Review by bresch — APPROVED** [review]

**Pull request merged**